src/Cli/dotnet/OptionForwardingExtensions.cs

```diff
@@ -21,8 +21,9 @@ public static class OptionForwardingExtensions
         public static ForwardedOption<string[]> ForwardAsProperty(this ForwardedOption<string[]> option) => option
             .SetForwardingFunction((optionVals) =>
                 optionVals
-                    .Select(optionVal => optionVal.Replace(";", "%3B")) // must escape semicolon-delimited property values when forwarding them to MSBuild
-                    .Select(optionVal => $"{option.Aliases.FirstOrDefault()}:{optionVal}")
+                    .SelectMany(Microsoft.DotNet.Cli.Utils.MSBuildPropertyParser.ParseProperties)
+                    // must escape semicolon-delimited property values when forwarding them to MSBuild
+                    .Select(optionVal => $"{option.Aliases.FirstOrDefault()}:{optionVal.key}={optionVal.value.Replace(";", "%3B")}")
```

Comment:
Is `;` the only value that needs to be encoded? And can `FirstOrDefault` ever return a `null` result here?

Reply:
Yeah, semicolon is the only thing we need to escape - it comes from the meaning of `:` in an MSbuild Item context as a collection delimiter :)

The FirstOrDefault will never throw a null here - it's grabbing the 'best' alias of the property option which is `--property`.